### PR TITLE
take care of hermes javascript engine

### PR DESCRIPTION
**Pull request by ice6**
It is sad that hermes do not support regex named group.

https://github.com/facebook/hermes/blob/main/doc/RegExp.md

same issue others encountered as me :)

https://github.com/facebook/hermes/issues/89

because this library is so foundamental and small, it will be a pleasure if this library can do the favor.

:)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -129,12 +129,12 @@ function assembleStyles() {
 		},
 		hexToRgb: {
 			value: hex => {
-				const matches = /(?<colorString>[a-f\d]{6}|[a-f\d]{3})/i.exec(hex.toString(16));
+				const matches = /([a-f\d]{6}|[a-f\d]{3})/i.exec(hex.toString(16));
```
Comment (Qix-) [inline index.js:132]:
```suggestion
				const matches = /[a-f\d]{6}|[a-f\d]{3}/i.exec(hex.toString(16));
```

**Review by Qix- — CHANGES_REQUESTED** [review]
I don't normally care to support non-compliant javascript engines but in this case the change makes sense. Named groups and destructuring is unnecessary here and only burns cycles anyway.

Just one change from me.

**Comment by sindresorhus** [comment]
While this is a seemingly simple change, it's a slippery slope. Hermes only supports ES6, so this issue will be hit again in the future.

**Comment by ice6** [comment]
> While this is a seemingly simple change, it's a slippery slope. Hermes only supports ES6, so this issue will be hit again in the future.

It will be ok to me if you add me as a maintainer of this little library. :)

**Comment by ice6** [comment]
for those suffering such little problem, now I recommend https://www.npmjs.com/package/patch-package.
as the package claimed, it is indeed a savior.

**Comment by bombillazo** [comment]
> for those suffering such little problem, now I recommend https://www.npmjs.com/package/patch-package. as the package claimed, it is indeed a savior.

What would be the solution using patch-package? What is the patch required to remove the named group?

**Pull request closed without merging**